core/src/main/java/org/sonarsource/sonarlint/core/analyzer/issue/DeprecatedIssueAdapterForFilter.java

```diff
@@ -0,0 +1,183 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.analyzer.issue;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import org.sonar.api.issue.Issue;
+import org.sonar.api.issue.IssueComment;
+import org.sonar.api.resources.Project;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.utils.Duration;
+
+/**
+ * @deprecated since 5.3
+ */
+@Deprecated
+class DeprecatedIssueAdapterForFilter implements Issue {
+  private final Project project;
+  private final DefaultClientIssue rawIssue;
+  private final String componentKey;
+
+  DeprecatedIssueAdapterForFilter(Project project, DefaultClientIssue rawIssue, String componentKey) {
+    this.project = project;
+    this.rawIssue = rawIssue;
+    this.componentKey = componentKey;
+  }
+
+  @Override
+  public String key() {
+    throw unsupported();
+  }
+
+  @Override
+  public String componentKey() {
+    return componentKey;
+  }
+
+  @Override
+  public RuleKey ruleKey() {
+    return RuleKey.parse(rawIssue.getRuleKey());
+  }
+
+  @Override
+  public String language() {
+    throw unsupported();
+  }
+
+  @Override
+  public String severity() {
+    return rawIssue.getSeverity();
+  }
+
+  @Override
+  public String message() {
+    return rawIssue.getMessage();
+  }
+
+  @Override
+  public Integer line() {
+    return rawIssue.getStartLine();
+  }
+
+  @Override
+  @Deprecated
+  public Double effortToFix() {
+    throw unsupported();
+  }
+
+  @Override
+  public String status() {
+    return Issue.STATUS_OPEN;
+  }
+
+  @Override
+  public String resolution() {
+    return null;
+  }
+
+  @Override
+  public String reporter() {
+    throw unsupported();
+  }
+
+  @Override
+  public String assignee() {
+    return null;
+  }
+
+  @Override
+  public Date creationDate() {
+    return project.getAnalysisDate();
+  }
+
+  @Override
+  public Date updateDate() {
+    return null;
+  }
+
+  @Override
+  public Date closeDate() {
+    return null;
+  }
+
+  @Override
+  public String attribute(String key) {
+    return attributes().get(key);
+  }
+
+  @Override
+  public Map<String, String> attributes() {
+    return Collections.emptyMap();
+  }
+
+  @Override
+  public String authorLogin() {
+    throw unsupported();
+  }
+
+  @Override
+  public String actionPlanKey() {
+    throw unsupported();
+  }
+
+  @Override
+  public List<IssueComment> comments() {
+    throw unsupported();
+  }
+
+  @Override
+  public boolean isNew() {
+    throw unsupported();
+  }
+
+  @Deprecated
+  @Override
+  public Duration debt() {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)